--- fabric-2.7.1.orig/fabric/tunnels.py	2021-12-04 20:24:42.000000000 -0300
+++ fabric-2.7.1/fabric/tunnels.py	2022-09-09 10:35:33.433738130 -0300
@@ -68,6 +68,9 @@
                 tun_sock, local_addr = sock.accept()
                 # Set TCP_NODELAY to match OpenSSH's forwarding socket behavior
                 tun_sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
+            except BlockingIOError:
+                time.sleep(0.01)
+                continue
             except socket.error as e:
                 if e.errno is errno.EAGAIN:
                     # TODO: make configurable
